src/api/aiSidebarManager.ts

```diff
@@ -0,0 +1,315 @@
+	 *
+	 * @param id            The prompt ID for which to perform the generate call.
+	 * @param writtenPrompt The prompt as written by a user.
+	 * @param context       The context provided for the prompt.
+	 */
+	private async _generateAndRegisterPromptResponse(
+		id: PromptId,
+		writtenPrompt: string,
+		context: string
+	): Promise<void> {
+		this._setState('loading');
+
+		// We don't need the IDs for the request, just the history.
+		const allAiHistoryWithoutId = this._allAiHistoryWithId.map(
+			(aiHistoryWithId) => aiHistoryWithId.history
+		);
+
+		try {
+			const response = await aiConnector.generate(
+				writtenPrompt,
+				allAiHistoryWithoutId,
+				context
+			);
+
+			const history: AiHistory = {
+				request: {
+					prompt: writtenPrompt,
+					contextContent: context,
+				},
+				response,
+			};
+
+			this._registerHistoryPrompt(id, history);
+			this._addResponseToAllAiHistory(id, history);
+
+			this.promptResponsesChangedNotifier.executeCallbacks();
+			this._setState('idle');
+		} catch (_error: unknown) {
+			this._setState('errored');
+		}
+	}
+
+	/**
+	 * Fires a generate request for a new prompt. Will use the current selection
+	 * as context for the given prompt.
+	 *
+	 * @param prompt The written prompt to send to the AI assistant.
+	 */
+	public async generateResponseForPrompt(
+		writtenPrompt: string
+	): Promise<void> {
+		const id = generateUUID();
+		const queryVariables = {
+			startContainer: selectionManager.getStartContainer(),
+			startOffset: selectionManager.getStartOffset(),
+			endContainer: selectionManager.getEndContainer(),
+			endOffset: selectionManager.getEndOffset(),
+		};
+		const promptContext = evaluateXPathToString(
+			xq`fonto:curated-text-in-range($startContainer, $startOffset, $endContainer, $endOffset)`,
+			null,
+			readOnlyBlueprint,
+			queryVariables
+		);
+
+		await this._generateAndRegisterPromptResponse(
+			id,
+			writtenPrompt,
+			promptContext
+		);
+	}
+
+	/**
+	 * "Retries" the prompt with the given ID by sending a new generate request
+	 * for that prompt. This will not update the context content or prompt.
+	 *
+	 * @param id The ID of the prompt to retry.
+	 */
+	public async generateResponseForExistingPrompt(
+		id: PromptId
+	): Promise<void> {
+		const promptObject = this._promptResponses.find(
+			(promptResponse) => promptResponse.id === id
+		);
+
+		const request = promptObject.history[0].request;
+		const writtenPrompt = request.prompt;
+		const promptContext = request.contextContent;
+
+		const initialPromptResponse = this._promptResponses.find(
+			(promptResponse) => promptResponse.id === id
+		);
+		if (!initialPromptResponse) {
+			this._setState('errored');
+		}
+
+		await this._generateAndRegisterPromptResponse(
+			id,
+			writtenPrompt,
+			promptContext
+		);
+	}
+
+	/**
+	 * Resets the conversation history. Returns the sidebar to its initial state.
+	 */
+	public resetConversation(): void {
```

Comment:
Should the "_state" also be reset in this case?

Reply:
Yes, that's correct.